src/js/structs/ServiceTree.js

```diff
@@ -91,6 +102,16 @@ module.exports = class ServiceTree extends Tree {
     });
   }
 
+  getInstances() {
```

Comment:
Let's do this refactoring in a different PR. The `getInstances` method only implements the current service struct interface and we should refactor both at a stroke.